apl/tabular-operators/join-operator.mdx

```diff
@@ -0,0 +1,175 @@
+
+### Choose the right kind of join
+
+- Use `inner` for standard joins where you need all matches.
+- Use `leftouter` or `rightouter` when you need to retain all rows from one table.
+- Use `leftanti` or `rightanti` to find rows that do not match.
+- Use `fullouter` for complete combinations of both tables.
+- Use `leftsemi` or `rightsemi` to filter rows based on existence in another table.
+
+## For users of other query languages
+
+If you come from other query languages, this section explains how to adjust your existing queries to achieve the same results in APL.
+
+<AccordionGroup>
+<Accordion title="Splunk SPL users">
+
+The `join` operator in APL works similarly to the `join` command in Splunk SPL. However, APL provides additional flexibility by supporting various join types (e.g., `inner`, `outer`, `leftouter`). Splunk uses a single default join type.
+
+<CodeGroup>
+```sql Splunk example
+index=logs | join type=inner [search index=traces]
+```
+
+```kusto APL equivalent
+['sample-http-logs']
+| join kind=inner ['otel-demo-traces'] on $left.id == $right.trace_id
+```
+</CodeGroup>
+
+</Accordion>
+<Accordion title="ANSI SQL users">
+
+The `join` operator in APL resembles SQL joins but uses distinct syntax. SQL uses `FROM` and `ON` clauses, whereas APL uses the `join` operator with explicit `kind` and `on` clauses.
+
+<CodeGroup>
+```sql SQL example
+SELECT *
+FROM logs
+JOIN traces
+ON logs.id = traces.trace_id
+```
+
+```kusto APL equivalent
+['sample-http-logs']
+| join kind=inner ['otel-demo-traces'] on $left.id == $right.trace_id
+```
+</CodeGroup>
+
+</Accordion>
+</AccordionGroup>
+
+## Usage
+
+### Syntax
+
+```kusto
+LeftDataset
+| join kind=KindOfJoin RightDataset on Conditions
+```
+
+### Parameters
+
+- `LeftDataset`: The first dataset, also known as the outer table or the left side of the join. If you expect one of the datasets to contain consistently less data than the other, specify the smaller dataset as the left side of the join.
+- `RightDataset`: The second dataset, also known as the inner table or the right side of the join.
+- `KindOfJoin`: Optionally, the [kind of join](#kinds-of-join) to perform. The only supported value and the default is `inner`.
+- `Conditions`: The conditions for matching rows. The conditions determine how Axiom matches rows from the `LeftDataset` with rows from the `RightDataset`.
+    - To join datasets on a field that has the same name in the two datasets, simply use the field name. For example, `on id`.
+    - To join datasets on a field that has different names in the two datasets, define the two field names in an equality expression such as `on $left.id == $right.trace_id`.
+    - The two sides of the equality expression must have the same data type.
+    - You can use expressions in the join conditions. For example, to compare two fields of different data types, use `on $left.id_string == tostring($right.trace_id_int)`.
```

Comment:
You can't use functions and `$left` and `$right`. It is only available if you want to specify fields. By default, left is the left side of the equals and right is the right. So you really only need them if you want to do `$right.right_id == $left.left_id`, which is diabolical :-)